pydantic_ai_slim/pydantic_ai/agent/__init__.py

```diff
@@ -243,6 +245,7 @@ def __init__(
         instrument: InstrumentationSettings | bool | None = None,
         history_processors: Sequence[HistoryProcessor[AgentDepsT]] | None = None,
         event_stream_handler: EventStreamHandler[AgentDepsT] | None = None,
+        ignore_warning_cost: bool = True,
```

Comment:
I changed this to True to make tests pass easily, since many things were failing where we didn't know the price.

I think that `Agent.run` itself shouldn't create a warning by default, but it should note when a price is unknown. The warning should be emitted when the user explicitly requests the cost of a run.

Reply:
Is the warning useful if it's opt-in to actually have it?

Reply:
the parameter wouldn't be here, it would always collect and store the information about future warnings during the run. there would be a parameter on some other method for retrieving the cost, where the warning is emitted by default.

Reply:
Makes sense